src/AutoRest.CSharp/Common/Output/Models/Types/ModelFactoryTypeProvider.cs

```diff
@@ -49,13 +113,13 @@ private static bool RequiresModelFactory(SchemaObjectType model)
                 return false;
             }
 
-            if (model.SerializationConstructor.Parameters.Any(p => !p.Type.IsPublic))
+            if (model.SerializationConstructor.Signature.Parameters.Any(p => !p.Type.IsPublic))
             {
                 return false;
             }
 
             return model.Constructors
-                .Where(c => c.Declaration.Accessibility == "public")
+                .Where(c => c.Signature.Modifiers == "public")
```

Comment:
nit: This is not a correct comparison if the property is for modifiers.

Reply:
I can change it to enum, but so far it can only be either `"public"` or `"internal"`.